strongbox-web-core/src/main/java/org/carlspring/strongbox/security/authentication/suppliers/NpmLoginAuthenticationSupplier.java

```diff
@@ -0,0 +1,83 @@
+package org.carlspring.strongbox.security.authentication.suppliers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.carlspring.strongbox.authentication.api.password.PasswordAuthentication;
+import org.carlspring.strongbox.controllers.layout.npm.NpmArtifactController;
+import org.carlspring.strongbox.controllers.layout.npm.NpmUser;
+
+import org.springframework.core.annotation.Order;
+import org.springframework.http.MediaType;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@Component
+@Order(4)
+public class NpmLoginAuthenticationSupplier
+        implements AuthenticationSupplier
+{
+
+    @Inject
+    private ObjectMapper objectMapper;
+
+    @CheckForNull
+    @Override
+    public Authentication supply(@Nonnull HttpServletRequest request)
+    {
+        NpmUser npmUser = deserializeNpmUser(request);
+
+        if (isValidNpmUser(npmUser))
+        {
+            return new PasswordAuthentication(npmUser.getName(), npmUser.getPassword());
+        }
+        else
+        {
+            throw new BadCredentialsException("invalid.credentials");
+        }
+
+    }
+
+    @Override
+    public boolean supports(@Nonnull HttpServletRequest request)
+    {
+        return "PUT".equalsIgnoreCase(request.getMethod()) &&
```

Comment:
@LeifBrooks we better check that this request is for repository with npm layout. you can do it similar to `NugetApiKeyAuthenticationSupplier`.

Reply:
So use the same exact code from  NugetApiKeyAuthenticationSupplier.supports()? Should we have them inherit it from a parent? or just copy for now

Reply:
@LeifBrooks: Could you use here `RequestMethod.PUT.name()` instead of the String constant?

Reply:
should we still include:
`request.getRequestURI().contains("/-/user/org.couchdb.user:")`
?

We can move the string to where constants are stored, but I think it may be needed. Otherwise how do we differentiate between a PUT that is just trying to do an "npm publish" and a call that is trying to do "npm login/adduser", which the authenticator is for?